lib/actions/user.actions.ts

```diff
@@ -72,3 +72,18 @@ export const deleteUserById = async ({ _id }: DeleteUserParams) => {
     console.log(error);
   }
 };
+export const updateUserSocials = async (socials: Partial<IUser["socials"]>) => {
+  try {
+    await dbConnect();
+    const session = await getSession();
+    const email = session?.user?.email;
+    const updatedUser = await User.findOneAndUpdate(
+      { email },
+      { socials },
+      { new: true }
+    );
+    return updatedUser;
```

Comment:
Returning the user here gives us an issue. Let's return something for user feedback (you have none at the moment). 

![image](https://github.com/agordon123/git_note/assets/4108484/6294edf3-ceb2-4cf4-831e-48ae503b6834)
(We had no reason to return this anyway as you currently aren't using a return value)

We can return true or false to mark a success or failure.

```js
export const updateUserSocials = async (socials: Partial<IUser["socials"]>) => {
  try {
    await dbConnect();
    const session = await getSession();
    const email = session?.user?.email;
    await User.findOneAndUpdate({ email }, { socials }, { new: true });
    return true;
  } catch (error) {
    console.log(error);
    return false;
  }
};
```

Now we return true if this is successful and false it if fails. So you can use that to give the user some feedback.

In your "onSubmit" for the `EditSocials` you can grab the return value to show some notification on success or failure.

```js
      const success = await updateUserSocials({
        twitter: {
          username: twitter?.username ?? "",
          url: twitter?.url ?? "",
        },
        instagram: {
          username: instagram?.username ?? "",
          url: instagram?.url ?? "",
        },
        linkedin: {
          username: linkedin?.username ?? "",
          url: linkedin?.url ?? "",
        },
        github: {
          username: github?.username ?? "",
          url: github?.url ?? "",
        },
        dribbble: {
          username: dribbble?.username ?? "",
          url: dribbble?.url ?? "",
        },
        facebook: {
          username: facebook?.username ?? "",
          url: facebook?.url ?? "",
        },
      });
      if (success) {
        // Handle success (socials updated)
      }else{
        // Handle failure (something went wrong try again)
      }
    } catch (error) {
      // Handle error
      // Handle failure (something went wrong, try again)
    }
    ```
    
    
   Please use a 'useTransition' hook as well (like you did in SignUpForm.tsx) so that we can show a loading state on the button and disable the button while the database query is happening)

Reply:
you got it.  didnt know about the resolve conversation thing i wasnt sure how to use that

Reply:
@agordon123 all good!